src/shell.ts

```diff
@@ -109,7 +111,43 @@ function execCore(cmd: string, opts: any): Promise<ShellResult> {
     });
 }
 
+export function shellEnvironment(baseEnvironment: any): any {
+    let env = Object.assign({}, baseEnvironment);
+    let pathVariable = pathVariableName(env);
+    for (const tool of ['kubectl', 'helm', 'draft']) {
+        const toolPath = vscode.workspace.getConfiguration('vs-kubernetes')[`vs-kubernetes.${tool}-path`];
+        if (toolPath) {
+            const toolDirectory = path.dirname(toolPath);
+            const currentPath = env[pathVariable];
+            env[pathVariable] = (currentPath ? `${currentPath}${pathEntrySeparator()}` : '') + toolDirectory;
+        }
+    }
+
+    const kubeconfig: string = vscode.workspace.getConfiguration('vs-kubernetes')['vs-kubernetes.kubeconfig'];
+    if (kubeconfig) {
+        env['KUBECONFIG'] = kubeconfig;
+    }
+
+    return env;
+}
+
+function pathVariableName(env: any): string {
+    if (isWindows) {
+        for (const v of Object.keys(env)) {
+            if (v.toLowerCase() === "path") {
+                return v;
```

Comment:
if we're iterating through this environment dict to check if a key named `path` exists, why not just return `"path"` immediately?

Reply:
Because JavaScript is case-sensitive, but the Windows environment isn't.  So if I return `path`, but the Windows environment variable is called `Path` (as it is in PowerShell), then the `env` object ends up with two properties `{ …; Path: "whatever;it;inherited;from;parent"; path: "kubectl-dir;helm-dir;draft-dir"; … }`, and then when `shelljs` sets the child process environment variables, the second one clobbers the first one.  Hence, I need to find the property of the `env` object which matches `path` _ignoring case_.

Ask me how I found this out...  **grin**

Reply:
hah! could you perhaps use `process.env.SHELL` to identify if the parent shell is powershell vs. windows command prompt?

Reply:
I could... or I could just not take a dependency on specific shells (or on the presumption that these shells guarantee a particular casing of PATH), and use whatever casing of PATH happens to be present in this environment.  This is safer, and probably no more complex once you allow for all the switches in the check-the-shell approach.

Reply:
On the plus side this discussion did get me to spot an error in this code and now I really need to find the `tslint` option for "die, implicit truthiness and falseness, die."